Alter Gulp tasks and compile patterns.

occupant opened the pull request:
Example to illustrate:

https://github.com/UBCCM/UBC-DEF/issues/13

occupant committed:
Add base type, drawer nav component (?) and hamburder (wip)

occupant committed:
Alter Gulp tasks and naming conventions for compiled scss

ADD a sample Humburger button and slidein drawer nav component with test files in the new 'test' directory.
CHANGE the main gulp styles task to allow multiple files of any name to be compiled from that directory.
ADD a new gulp task (components) to allow components to be compiled separately. This deprecates the minify task (and maybe best renamed as such)

richardtape commented on dist/css/styles-minimal.dev.css:
```diff
@@ -405,27 +405,132 @@ template {
 [hidden] {
   display: none; }
```
Are we all happy with the pattern of having the closing curly brace on the last line? Is there a reason for this other than just a stylistic choice?

occupant replied:
No, that's wrong - I didn't lint the code! I'll add that task to the travis tasks.

richardtape replied:
I'm so very glad to hear that because I hate it so much ;)

occupant replied:
Bad news. It has nothing to do with the lint task - it's the compile task. The lint enforces the source to be good, but the compile has its own ideas. I'm going to ignore this for the time being, but it's something that we should revisit.

richardtape commented on src/sass/base/_typography.scss:
```diff
@@ -1,3 +1,173 @@
+
+// From Typebase.scss v0.1.0 | https://github.com/devinhunt/typebase.css/blob/master/src/typebase.scss
+
+// Typesetting variables.
```
Are we going to keep variables compartmentalized within components or have a central variables file? My thinking; in a component, pull in the centralized value if it exists, otherwise revert to a default set in the component. Thoughts?

occupant replied:
Yes, this is what I was thinking. After having used the 'one file to rule them all' method, I appreciate grouping my variables with where they're in use.

hazuremon replied:
I agree to pull in from centralized value (will this be a centralized file then?) and then have default set in component as well.

richardtape commented on src/sass/styles-minimal.scss:
```diff
@@ -0,0 +1,18 @@
+@charset 'UTF-8';
```
Just to mention as per issue #13 that I suggest this file be called `styles-critical.scss` (with the appropriate compiled version name)

occupant replied:
Agreed

richardtape commented on tests/typography/index.html:
```diff
@@ -0,0 +1,68 @@
+    /* https://basehold.it - baseline bg image - the 24 denotes the line height - change if needed */
+    
+    .grid-wrapper {
+      background-image: url('http://basehold.it/i/24');
```
I think we should use https assets in demos when they're available.

This redirects to the https version anyway, but let's be explicit.

occupant replied:
Fixed

richardtape reviewed (commented):
Have made various comments and asked some questions

occupant committed:
Update Gulp Tasks and output

ADD editorconfig
UPDATE - code formatting (to reflect editorconfig settings)
UPDATE - sass-lint rules
UPDATE - linted all SCSS
UPDATE - moved mapped css to dev directory
UPDATE - renamed 'component' task to 'minify', removed old minify task
UPDATE - renamed styles-minimal to styles-critical

occupant committed:
UPDATE travis ci tasks  to include 'lint'

occupant committed:
Merge branch 'dev' into typography

occupant committed:
Merge branch 'dev' into typography

occupant committed:
UPDATE variable lint rule to follow bem format

occupant committed:
UPDATE color definitions to hsla format, added css3 variables to variables.scss file

occupant committed:
UPDATE component variable names, test document links

occupant committed:
UPDATE Changed hsla css variable to hex variable as a workaround for bug in sasslint

occupant committed:
test

occupant committed:
Remove sassdoc

occupant committed:
css fixed with sass-lint

occupant committed:
Remove sassdoc from gulp file

occupant committed:
Update gulp to 4.x and all gulp plugins to latest version

occupant committed:
Add watch task back in

occupant committed:
update styles

occupant committed:
Updates

occupant committed:
Adjust type and nav anchors, decrease autoprefix threshhold to last 4

occupant committed:
Fix watch task

occupant committed:
update node to v.8 in travis tests
